node/flows.re

```diff
@@ -246,19 +248,32 @@ let try_to_commit_state_hash = (~prev_validators, state, block, signatures) => {
   });
 };
 
+let hash_new_state_root = (state, protocol, update_state) => {
+  let task_pool = get_task_pool^();
+  let (snapshot_ref, snapshots) =
+    Snapshots.add_snapshot_ref(
+      ~block_height=protocol.block_height,
+      state.Node.snapshots,
+    );
+  let _ =
```

Comment:
with async stuff is very easy to miss a promise or a domain by accident, at least do `(_task: type)`

Reply:
Done